vote/src/vote_state_view.rs

```diff
@@ -0,0 +1,506 @@
+use {
+    self::{
+        field_frames::{
+            AuthorizedVotersListFrame, EpochCreditsItem, EpochCreditsListFrame, RootSlotFrame,
+            RootSlotView, VotesFrame,
+        },
+        frame_v1_14_11::VoteStateFrameV1_14_11,
+        frame_v3::VoteStateFrameV3,
+        list_view::ListView,
+    },
+    core::fmt::Debug,
+    solana_clock::{Epoch, Slot},
+    solana_pubkey::Pubkey,
+    solana_vote_interface::state::{BlockTimestamp, Lockout},
+    std::sync::Arc,
+};
+#[cfg(feature = "dev-context-only-utils")]
+use {
+    bincode,
+    solana_vote_interface::state::{VoteState, VoteStateVersions},
+};
+
+mod field_frames;
+mod frame_v1_14_11;
+mod frame_v3;
+mod list_view;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum VoteStateViewError {
+    AccountDataTooSmall,
+    InvalidVotesLength,
+    InvalidRootSlotOption,
+    InvalidAuthorizedVotersLength,
+    InvalidEpochCreditsLength,
+    OldVersion,
+    UnsupportedVersion,
+}
+
+pub type Result<T> = core::result::Result<T, VoteStateViewError>;
+
+enum Field {
+    NodePubkey,
+    Commission,
+    Votes,
+    RootSlot,
+    AuthorizedVoters,
+    EpochCredits,
+    LastTimestamp,
+}
+
+/// A view into a serialized VoteState.
+///
+/// This struct provides access to the VoteState data without
+/// deserializing it. This is done by parsing and caching metadata
+/// about the layout of the serialized VoteState.
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "frozen-abi", derive(AbiExample))]
```

Comment:
why do we need frozen-abi?

Reply:
It's just that this struct is now part of `VoteAccountInner` and afaik there's no way to skip a field for the `AbiExample` derivation